.github/workflows/sycl_mlir_weekly.yml

```diff
@@ -0,0 +1,89 @@
+name: SYCL MLIR Weekly
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 22 * * THU'
+
+jobs:
+  ubuntu2204_build:
+    if: github.repository == 'intel/llvm'
+    uses: ./.github/workflows/sycl_linux_build.yml
+    secrets: inherit
+    with:
+      build_ref: sycl-mlir
+      build_cache_root: "/__w/"
```

Comment:
This is bad because it will never hit the cache and will evict normal origin/sycl builds from the cache.

Reply:
I'm not sure about the second statement, but, after my testing, I inferred we were getting cache hits. Compare build times of [this action](https://github.com/intel/llvm/actions/runs/6026431874) vs [this one](https://github.com/intel/llvm/actions/runs/6027561718).

If the second is a concern to you, we may want to revisit "SYCL MLIR Pre Commit on Linux" workflow too, as it is using the same value ([see](https://github.com/intel/llvm/blob/5c6fb36c2068ae91e34b2ee2ade3f02f45605a50/.github/workflows/sycl_mlir_precommit_linux.yml#L64)).

Reply:
Yes, the second is the concern. What I meant by miss is that I expected this to run once a week and then the cache would contain normal SYCL builds without MLIR. Your comparison isn't accurate though. One is using our gen9 runner, another is using more powerful amd runner.